src/main/java/com/minecolonies/coremod/entity/ai/minimal/EntityAIInteractToggleAble.java

```diff
@@ -0,0 +1,254 @@
+package com.minecolonies.coremod.entity.ai.minimal;
+
+import net.minecraft.block.*;
+import net.minecraft.entity.MobEntity;
+import net.minecraft.entity.ai.goal.Goal;
+import net.minecraft.pathfinding.GroundPathNavigator;
+import net.minecraft.pathfinding.Path;
+import net.minecraft.pathfinding.PathPoint;
+import net.minecraft.state.properties.BlockStateProperties;
+import net.minecraft.util.math.BlockPos;
+import org.jetbrains.annotations.NotNull;
+
+import java.util.*;
+import java.util.function.Predicate;
+
+/**
+ * AI Task for toggling blocks open/closed when collided
+ */
+public class EntityAIInteractToggleAble extends Goal
+{
+    /**
+     * Number of blocks to check for the fence gate - height.
+     */
+    private static final int HEIGHT_TO_CHECK = 2;
+
+    /**
+     * Number of blocks to check for the fence gate - length.
+     */
+    private static final int LENGTH_TO_CHECK = 2;
+
+    /**
+     * The min distance the gate has to be from the citizen.
+     */
+    private static final double MIN_DISTANCE = 2.25D;
+
+    /**
+     * Default predicate for all 3 vanilla blocks
+     */
+    public static final Predicate<BlockState> GATE_DOOR_TRAP = new Predicate<BlockState>()
+    {
+        @Override
+        public boolean test(final BlockState state)
+        {
+            final Block block = state.getBlock();
+            return block instanceof DoorBlock || block instanceof FenceGateBlock || block instanceof TrapDoorBlock;
+        }
+    };
+
+    /**
+     * Only trapdoors and fence gates
+     */
+    public static final Predicate<BlockState> GATE_TRAP = new Predicate<BlockState>()
+    {
+        @Override
+        public boolean test(final BlockState state)
+        {
+            final Block block = state.getBlock();
+            return block instanceof DoorBlock || block instanceof FenceGateBlock || block instanceof TrapDoorBlock;
+        }
```

Comment:
Why aren't those just <Block> predicates if we don't need the state anyway?

Reply:
"Block predicate" I mean

Reply:
wanted to give them the chance to also add state property conditions, like state.getProperty(OPEN) == true